src/engine/DebugOverlay.EntityLabel.cs

```diff
@@ -0,0 +1,158 @@
+﻿using System.Collections.Generic;
+using Godot;
+
+/// <summary>
+///   Partial class: Entity label
+/// </summary>
+public partial class DebugOverlay
+{
+    private readonly Dictionary<RigidBody, Label> entityLabels = new();
+
+    private bool showEntityLabels;
+    private Font smallerFont = null!;
+    private Camera? activeCamera;
+
+    private bool ShowEntityLabels
+    {
+        get => showEntityLabels;
+        set
+        {
+            showEntityLabels = value;
+            labelsLayer.Visible = value;
+        }
+    }
+
+    private void EntityLabelReady()
+    {
+        labelsLayer = GetNode<Control>(EntityLabelsPath);
+        smallerFont = GD.Load<Font>("res://src/gui_common/fonts/Lato-Regular-Tiny.tres");
+    }
+
+    private void EntityLabelEnterTree()
+    {
+        var rootTree = GetTree();
+        rootTree.Connect("node_added", this, nameof(OnNodeAdded));
+        rootTree.Connect("node_removed", this, nameof(OnNodeRemoved));
+    }
+
+    private void EntityLabelExitTree()
+    {
+        var rootTree = GetTree();
+        rootTree.Disconnect("node_added", this, nameof(OnNodeAdded));
+        rootTree.Disconnect("node_removed", this, nameof(OnNodeRemoved));
+    }
+
+    private void EntityLabelProcess()
+    {
+        if (activeCamera is not { Current: true })
+            activeCamera = GetViewport().GetCamera();
+
+        if (showEntityLabels)
+        {
+            UpdateEntityLabels();
+        }
+    }
```

Comment:
I think these can just be in the main class without the need for separate methods. The microbe class for example already handles all its init for its partial classes in the main class.

Reply:
I think it looks better if I can find all relevant logics in a single file. Being a simple function, the compiler may as well merge this into the main one.

Reply:
It looks more tangled to me having these in the partial classes. In my view, only the distinct logics should be in the partial classes. EnterTree, ExitTree, Ready or Process logics are not distinct and is associated with the DebugOverlay node itself should be kept in the main DebugOverlay class where it makes sense.